src/shared/formatters/red4ext-rs.formatter.ts

```diff
@@ -0,0 +1,95 @@
+import { RedClassAst } from '../red-ast/red-class.ast';
+import { RedFunctionAst } from '../red-ast/red-function.ast';
+import { RedTypeAst } from '../red-ast/red-type.ast';
+import { CodeSyntax } from '../services/settings.service';
+import { CodeFormatter, CodeVariableFormat } from './formatter';
+
+export class Red4extRsFormatter extends CodeFormatter {
+  constructor() {
+    super(true);
+  }
+  override formatPrototype(func: RedFunctionAst): string {
+    const args: string = this.formatPrototypeArguments(func);
+    const returnType: string = this.formatType(func.returnType);
+    const annotation: string = func.isNative
+      ? '#[redscript_global(native)]'
+      : '#[redscript_global]';
+
+    return `${annotation}\npub fn ${this.camelToSnakeCase(
+      func.name
+    )}(${args}) -> ${returnType}`;
+  }
+
+  override formatSpecial(
+    type: string,
+    func: RedFunctionAst,
+    memberOf?: RedClassAst
+  ): string {
+    return 'TODO';
+  }
+
+  protected override formatSelf(
+    func: RedFunctionAst,
+    memberOf?: RedClassAst
+  ): CodeVariableFormat | undefined {
+    return undefined;
+  }
+
+  protected override formatReturn(
+    func: RedFunctionAst
+  ): CodeVariableFormat | undefined {
+    return undefined;
+  }
+
+  protected override formatArguments(
+    func: RedFunctionAst,
+    selfName?: string
+  ): CodeVariableFormat[] {
+    return [];
+  }
+
+  protected override formatMemberStaticCall(
+    func: RedFunctionAst,
+    memberOf: RedClassAst
+  ): string {
+    return 'TODO';
+  }
+
+  protected override formatMemberCall(
+    selfVar: CodeVariableFormat,
+    func: RedFunctionAst
+  ): string {
+    return 'TODO';
+  }
+
+  protected override formatStaticCall(func: RedFunctionAst): string {
+    return 'TODO';
+  }
+
+  private formatPrototypeArguments(func: RedFunctionAst): string {
+    return func.arguments
+      .map(
+        (arg) =>
+          `${this.camelToSnakeCase(arg.name)}: ${this.formatType(arg.type)}`
+      )
+      .join(', ');
+  }
+
+  private formatCallArguments(func: RedFunctionAst): string {
+    return func.arguments.map((arg) => arg.name).join(', ');
+  }
+
+  private formatType(type?: RedTypeAst): string {
+    return type ? RedTypeAst.toString(type, CodeSyntax.rustRED4ext) : '()';
+  }
+
+  private formatFlags(func: RedFunctionAst): string {
+    return 'TODO';
+  }
+
+  private camelToSnakeCase(str: string) {
```

Comment:
It should be defined as a global helper in [string.ts](https://github.com/rayshader/cp2077-nativedb/blob/master/src/shared/string.ts). In case feature need to be reused elsewhere.

Return type is missing, usually all types are declared, except for args/return type of lambdas for example.

Reply:
Indeed! Done in 3461039.